pkg/restclient/peer.go

```diff
@@ -0,0 +1,27 @@
+package restclient
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"github.com/gluster/glusterd2/pkg/api"
+)
+
+// PeerProbe adds a peer to the Cluster
+func (c *RESTClient) PeerProbe(host string) error {
+	peerAddReq := api.PeerAddReq{
+		Addresses: []string{host},
+	}
+	reqBody, err := json.Marshal(peerAddReq)
+	if err != nil {
+		return err
+	}
+	return httpRESTAction("POST", c.baseURL+"/v1/peers", strings.NewReader(string(reqBody)), 201)
```

Comment:
All points makes sense.

> You can avoid passing the expected code, you can have a map of action and expected code and use it instead.

Expected code may differ from request to request, keeping map of action is not generic. Plugins restclient may expect different status codes, which can't be added to map.